### ISSUE TITLE
Doesn't detect new files?

### ISSUE THREAD

**Issue by chrisabrams** [substack/watchify#56]
Is there a configuration needed to get watchify to recognize new files?

**Comment by simonzack** [comment]
+1 to this, most of the gulp incremental build packages offer this.

**Comment by simonzack** [comment]
@aearly watchify uses chokidar which does support added files but it apparently doesn't use this functionality.

**Comment by aearly** [comment]
@simonzack the issue is glob expansion before the list of files is handed to watchify.

`watchify test/*.test.js -o public/js/tests.js`

gets expanded to:

`watchify test/foo.test.js test/bar.test.js test/baz.test.js -o public/js/tests.js`

 and then executed.

There is no way for watchify to know that `test/qux.test.js` has been added because it has no knowledge of the glob -- as far as it knows, you passed in a list of files.

**Comment by simonzack** [comment]
`watchify 'test/*.test.js' -o public/js/tests.js` could be used instead.

**Comment by aearly** [comment]
Yep, or probably something like `watchify --entry-glob 'test/*.test.js' -o public/js/tests.js` to make it explicit.
